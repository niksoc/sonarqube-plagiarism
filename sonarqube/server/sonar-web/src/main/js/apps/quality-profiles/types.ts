/*
 * SonarQube
 * Copyright (C) 2009-2020 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { Profile as BaseProfile } from '../../api/quality-profiles';

export interface Profile extends BaseProfile {
  depth: number;
  childrenCount: number;
}

export interface Exporter {
  key: string;
  name: string;
  languages: string[];
}

export interface ProfileChangelogEvent {
  action: string;
  authorName: string;
  date: string;
  params?: T.Dict<string | null>;
  ruleKey: string;
  ruleName: string;
}
