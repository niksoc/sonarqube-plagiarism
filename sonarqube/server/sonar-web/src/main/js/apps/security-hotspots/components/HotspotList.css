/*
 * SonarQube
 * Copyright (C) 2009-2020 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
.hotspot-list-header {
  padding: calc(2 * var(--gridSize)) var(--gridSize);
}

.hotspot-risk-header {
  padding: var(--gridSize);
}

.hotspot-category {
  background-color: white;
  border: 1px solid var(--barBorderColor);
}

.hotspot-category .hotspot-category-header {
  padding: calc(2 * var(--gridSize)) var(--gridSize);
  color: var(--baseFontColor);
  border-bottom: none;
  border-left: 4px solid;
}

.hotspot-category .hotspot-category-header:hover,
.hotspot-category .hotspot-category-header.contains-selected-hotspot {
  color: var(--blue);
}

.hotspot-category.HIGH .hotspot-category-header {
  border-left-color: var(--red);
}

.hotspot-category.MEDIUM .hotspot-category-header {
  border-left-color: var(--orange);
}

.hotspot-category.LOW .hotspot-category-header {
  border-left-color: var(--yellow);
}

.hotspot-item {
  color: var(--baseFontColor);
  display: block;
  padding: var(--gridSize) calc(2 * var(--gridSize));
  border: 1px solid transparent;
  border-top-color: var(--barBorderColor);
  transition: padding 0s, border 0s;
}

.hotspot-item:hover {
  background-color: var(--veryLightBlue);
  border: 1px dashed var(--blue);
  color: var(--baseFontColor);
}

.hotspot-item.highlight {
  background-color: var(--veryLightBlue);
  color: var(--baseFontColor);
  border: 1px solid var(--blue);
  cursor: unset;
}

.hotspot-risk-badge {
  color: white;
  text-transform: uppercase;
  display: inline-block;
  text-align: center;
  min-width: 48px;
  padding: 0 var(--gridSize);
  font-weight: bold;
  border-radius: 2px;
}

.hotspot-risk-badge.HIGH {
  background-color: var(--red);
}
.hotspot-risk-badge.MEDIUM {
  background-color: var(--orange);
}
.hotspot-risk-badge.LOW {
  background-color: var(--yellow);
}
