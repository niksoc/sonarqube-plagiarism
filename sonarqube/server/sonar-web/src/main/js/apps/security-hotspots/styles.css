/*
 * SonarQube
 * Copyright (C) 2009-2020 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
#security_hotspots .wrapper {
  position: fixed;
  /* top is defined programatically */
  bottom: 0;
  width: 100%;
}

#security_hotspots .layout-page {
  margin: 0 auto;
  min-width: var(--minPageWidth);
  max-width: 1280px;
  height: 100%;
}

#security_hotspots .filter-bar {
  max-width: 1280px;
  box-sizing: border-box;
  margin: 0 auto;
  padding: calc(2 * var(--gridSize)) var(--gridSize);
  border-bottom: 1px solid var(--barBorderColor);
}

#security_hotspots .sidebar {
  flex: 0 0 300px;
  border-right: 1px solid var(--barBorderColor);
  height: 100%;
  overflow-y: auto;
}

#security_hotspots .main {
  flex: 1 1 auto;
  background-color: white;
  /* Force flex to take parent width. */
  overflow-x: auto;
}

#security_hotspots .main .hotspot-content {
  overflow-y: auto;
  height: 100%;
  box-sizing: border-box;
}

.invisible {
  visibility: hidden;
}
