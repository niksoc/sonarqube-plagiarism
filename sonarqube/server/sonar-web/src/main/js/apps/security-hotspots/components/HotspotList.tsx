/*
 * SonarQube
 * Copyright (C) 2009-2020 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as classNames from 'classnames';
import { groupBy } from 'lodash';
import * as React from 'react';
import ListFooter from 'sonar-ui-common/components/controls/ListFooter';
import SecurityHotspotIcon from 'sonar-ui-common/components/icons/SecurityHotspotIcon';
import { translate, translateWithParameters } from 'sonar-ui-common/helpers/l10n';
import { HotspotStatusFilter, RawHotspot, RiskExposure } from '../../../types/security-hotspots';
import { groupByCategory, RISK_EXPOSURE_LEVELS } from '../utils';
import HotspotCategory from './HotspotCategory';
import './HotspotList.css';

interface Props {
  hotspots: RawHotspot[];
  hotspotsTotal?: number;
  isStaticListOfHotspots: boolean;
  loadingMore: boolean;
  onHotspotClick: (hotspot: RawHotspot) => void;
  onLoadMore: () => void;
  securityCategories: T.StandardSecurityCategories;
  selectedHotspot: RawHotspot;
  statusFilter: HotspotStatusFilter;
}

interface State {
  expandedCategories: T.Dict<boolean>;
  groupedHotspots: Array<{
    risk: RiskExposure;
    categories: Array<{ key: string; hotspots: RawHotspot[]; title: string }>;
  }>;
}

export default class HotspotList extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);

    this.state = {
      expandedCategories: { [props.selectedHotspot.securityCategory]: true },
      groupedHotspots: this.groupHotspots(props.hotspots, props.securityCategories)
    };
  }

  componentDidUpdate(prevProps: Props) {
    // Force open the category of selected hotspot
    if (
      this.props.selectedHotspot.securityCategory !== prevProps.selectedHotspot.securityCategory
    ) {
      this.handleToggleCategory(this.props.selectedHotspot.securityCategory, true);
    }

    // Compute the hotspot tree from the list
    if (
      this.props.hotspots !== prevProps.hotspots ||
      this.props.securityCategories !== prevProps.securityCategories
    ) {
      const groupedHotspots = this.groupHotspots(
        this.props.hotspots,
        this.props.securityCategories
      );
      this.setState({ groupedHotspots });
    }
  }

  groupHotspots = (hotspots: RawHotspot[], securityCategories: T.StandardSecurityCategories) => {
    const risks = groupBy(hotspots, h => h.vulnerabilityProbability);

    return RISK_EXPOSURE_LEVELS.map(risk => ({
      risk,
      categories: groupByCategory(risks[risk], securityCategories)
    })).filter(risk => risk.categories.length > 0);
  };

  handleToggleCategory = (categoryKey: string, value: boolean) => {
    this.setState(({ expandedCategories }) => ({
      expandedCategories: { ...expandedCategories, [categoryKey]: value }
    }));
  };

  render() {
    const {
      hotspots,
      hotspotsTotal,
      isStaticListOfHotspots,
      loadingMore,
      selectedHotspot,
      statusFilter
    } = this.props;

    const { expandedCategories, groupedHotspots } = this.state;

    return (
      <div className="huge-spacer-bottom">
        <h1 className="hotspot-list-header bordered-bottom">
          <SecurityHotspotIcon className="spacer-right" />
          {translateWithParameters(
            isStaticListOfHotspots ? 'hotspots.list_title' : `hotspots.list_title.${statusFilter}`,
            hotspots.length
          )}
        </h1>
        <ul className="big-spacer-bottom">
          {groupedHotspots.map(riskGroup => (
            <li className="big-spacer-bottom" key={riskGroup.risk}>
              <div className="hotspot-risk-header little-spacer-left">
                <span>{translate('hotspots.risk_exposure')}:</span>
                <div className={classNames('hotspot-risk-badge', 'spacer-left', riskGroup.risk)}>
                  {translate('risk_exposure', riskGroup.risk)}
                </div>
              </div>
              <ul>
                {riskGroup.categories.map(cat => (
                  <li className="spacer-bottom" key={cat.key}>
                    <HotspotCategory
                      categoryKey={cat.key}
                      expanded={expandedCategories[cat.key]}
                      hotspots={cat.hotspots}
                      onHotspotClick={this.props.onHotspotClick}
                      onToggleExpand={this.handleToggleCategory}
                      selectedHotspot={selectedHotspot}
                      title={cat.title}
                    />
                  </li>
                ))}
              </ul>
            </li>
          ))}
        </ul>
        <ListFooter
          count={hotspots.length}
          loadMore={!loadingMore ? this.props.onLoadMore : undefined}
          loading={loadingMore}
          total={hotspotsTotal}
        />
      </div>
    );
  }
}
