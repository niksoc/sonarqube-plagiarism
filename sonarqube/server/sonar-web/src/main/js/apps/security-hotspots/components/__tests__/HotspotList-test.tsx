/*
 * SonarQube
 * Copyright (C) 2009-2020 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { shallow } from 'enzyme';
import * as React from 'react';
import { mockRawHotspot } from '../../../../helpers/mocks/security-hotspots';
import { HotspotStatusFilter, RiskExposure } from '../../../../types/security-hotspots';
import HotspotList from '../HotspotList';

it('should render correctly', () => {
  expect(shallowRender()).toMatchSnapshot();
  expect(shallowRender({ loadingMore: true })).toMatchSnapshot();
});

it('should render correctly when the list of hotspot is static', () => {
  expect(shallowRender({ isStaticListOfHotspots: true })).toMatchSnapshot();
});

const hotspots = [
  mockRawHotspot({ key: 'h1', securityCategory: 'cat2' }),
  mockRawHotspot({ key: 'h2', securityCategory: 'cat1' }),
  mockRawHotspot({
    key: 'h3',
    securityCategory: 'cat1',
    vulnerabilityProbability: RiskExposure.MEDIUM
  }),
  mockRawHotspot({
    key: 'h4',
    securityCategory: 'cat1',
    vulnerabilityProbability: RiskExposure.MEDIUM
  }),
  mockRawHotspot({
    key: 'h5',
    securityCategory: 'cat2',
    vulnerabilityProbability: RiskExposure.MEDIUM
  })
];

it('should render correctly with hotspots', () => {
  expect(shallowRender({ hotspots })).toMatchSnapshot('no pagination');
  expect(shallowRender({ hotspots, hotspotsTotal: 7 })).toMatchSnapshot('pagination');
});

it('should update expanded categories correctly', () => {
  const wrapper = shallowRender({ hotspots, selectedHotspot: hotspots[0] });

  expect(wrapper.state().expandedCategories).toEqual({ cat2: true });

  wrapper.setProps({ selectedHotspot: hotspots[1] });

  expect(wrapper.state().expandedCategories).toEqual({ cat1: true, cat2: true });
});

it('should update grouped hotspots when the list changes', () => {
  const wrapper = shallowRender({ hotspots, selectedHotspot: hotspots[0] });

  wrapper.setProps({ hotspots: [mockRawHotspot()] });

  expect(wrapper.state().groupedHotspots).toHaveLength(1);
  expect(wrapper.state().groupedHotspots[0].categories).toHaveLength(1);
  expect(wrapper.state().groupedHotspots[0].categories[0].hotspots).toHaveLength(1);
});

function shallowRender(props: Partial<HotspotList['props']> = {}) {
  return shallow<HotspotList>(
    <HotspotList
      hotspots={[]}
      isStaticListOfHotspots={false}
      loadingMore={false}
      onHotspotClick={jest.fn()}
      onLoadMore={jest.fn()}
      securityCategories={{}}
      selectedHotspot={mockRawHotspot({ key: 'h2' })}
      statusFilter={HotspotStatusFilter.TO_REVIEW}
      {...props}
    />
  );
}
