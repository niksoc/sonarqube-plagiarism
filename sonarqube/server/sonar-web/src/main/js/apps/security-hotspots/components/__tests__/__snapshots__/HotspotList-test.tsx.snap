// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="huge-spacer-bottom"
>
  <h1
    className="hotspot-list-header bordered-bottom"
  >
    <SecurityHotspotIcon
      className="spacer-right"
    />
    hotspots.list_title.TO_REVIEW.0
  </h1>
  <ul
    className="big-spacer-bottom"
  />
  <ListFooter
    count={0}
    loadMore={[MockFunction]}
    loading={false}
  />
</div>
`;

exports[`should render correctly 2`] = `
<div
  className="huge-spacer-bottom"
>
  <h1
    className="hotspot-list-header bordered-bottom"
  >
    <SecurityHotspotIcon
      className="spacer-right"
    />
    hotspots.list_title.TO_REVIEW.0
  </h1>
  <ul
    className="big-spacer-bottom"
  />
  <ListFooter
    count={0}
    loading={true}
  />
</div>
`;

exports[`should render correctly when the list of hotspot is static 1`] = `
<div
  className="huge-spacer-bottom"
>
  <h1
    className="hotspot-list-header bordered-bottom"
  >
    <SecurityHotspotIcon
      className="spacer-right"
    />
    hotspots.list_title.0
  </h1>
  <ul
    className="big-spacer-bottom"
  />
  <ListFooter
    count={0}
    loadMore={[MockFunction]}
    loading={false}
  />
</div>
`;

exports[`should render correctly with hotspots: no pagination 1`] = `
<div
  className="huge-spacer-bottom"
>
  <h1
    className="hotspot-list-header bordered-bottom"
  >
    <SecurityHotspotIcon
      className="spacer-right"
    />
    hotspots.list_title.TO_REVIEW.5
  </h1>
  <ul
    className="big-spacer-bottom"
  >
    <li
      className="big-spacer-bottom"
      key="HIGH"
    >
      <div
        className="hotspot-risk-header little-spacer-left"
      >
        <span>
          hotspots.risk_exposure
          :
        </span>
        <div
          className="hotspot-risk-badge spacer-left HIGH"
        >
          risk_exposure.HIGH
        </div>
      </div>
      <ul>
        <li
          className="spacer-bottom"
          key="cat2"
        >
          <HotspotCategory
            categoryKey="cat2"
            hotspots={
              Array [
                Object {
                  "author": "Developer 1",
                  "component": "com.github.kevinsawicki:http-request:com.github.kevinsawicki.http.HttpRequest",
                  "creationDate": "2013-05-13T17:55:39+0200",
                  "key": "h1",
                  "line": 81,
                  "message": "'3' is a magic number.",
                  "project": "com.github.kevinsawicki:http-request",
                  "resolution": undefined,
                  "rule": "checkstyle:com.puppycrawl.tools.checkstyle.checks.coding.MagicNumberCheck",
                  "securityCategory": "cat2",
                  "status": "TO_REVIEW",
                  "updateDate": "2013-05-13T17:55:39+0200",
                  "vulnerabilityProbability": "HIGH",
                },
              ]
            }
            onHotspotClick={[MockFunction]}
            onToggleExpand={[Function]}
            selectedHotspot={
              Object {
                "author": "Developer 1",
                "component": "com.github.kevinsawicki:http-request:com.github.kevinsawicki.http.HttpRequest",
                "creationDate": "2013-05-13T17:55:39+0200",
                "key": "h2",
                "line": 81,
                "message": "'3' is a magic number.",
                "project": "com.github.kevinsawicki:http-request",
                "resolution": undefined,
                "rule": "checkstyle:com.puppycrawl.tools.checkstyle.checks.coding.MagicNumberCheck",
                "securityCategory": "command-injection",
                "status": "TO_REVIEW",
                "updateDate": "2013-05-13T17:55:39+0200",
                "vulnerabilityProbability": "HIGH",
              }
            }
            title="cat2"
          />
        </li>
        <li
          className="spacer-bottom"
          key="cat1"
        >
          <HotspotCategory
            categoryKey="cat1"
            hotspots={
              Array [
                Object {
                  "author": "Developer 1",
                  "component": "com.github.kevinsawicki:http-request:com.github.kevinsawicki.http.HttpRequest",
                  "creationDate": "2013-05-13T17:55:39+0200",
                  "key": "h2",
                  "line": 81,
                  "message": "'3' is a magic number.",
                  "project": "com.github.kevinsawicki:http-request",
                  "resolution": undefined,
                  "rule": "checkstyle:com.puppycrawl.tools.checkstyle.checks.coding.MagicNumberCheck",
                  "securityCategory": "cat1",
                  "status": "TO_REVIEW",
                  "updateDate": "2013-05-13T17:55:39+0200",
                  "vulnerabilityProbability": "HIGH",
                },
              ]
            }
            onHotspotClick={[MockFunction]}
            onToggleExpand={[Function]}
            selectedHotspot={
              Object {
                "author": "Developer 1",
                "component": "com.github.kevinsawicki:http-request:com.github.kevinsawicki.http.HttpRequest",
                "creationDate": "2013-05-13T17:55:39+0200",
                "key": "h2",
                "line": 81,
                "message": "'3' is a magic number.",
                "project": "com.github.kevinsawicki:http-request",
                "resolution": undefined,
                "rule": "checkstyle:com.puppycrawl.tools.checkstyle.checks.coding.MagicNumberCheck",
                "securityCategory": "command-injection",
                "status": "TO_REVIEW",
                "updateDate": "2013-05-13T17:55:39+0200",
                "vulnerabilityProbability": "HIGH",
              }
            }
            title="cat1"
          />
        </li>
      </ul>
    </li>
    <li
      className="big-spacer-bottom"
      key="MEDIUM"
    >
      <div
        className="hotspot-risk-header little-spacer-left"
      >
        <span>
          hotspots.risk_exposure
          :
        </span>
        <div
          className="hotspot-risk-badge spacer-left MEDIUM"
        >
          risk_exposure.MEDIUM
        </div>
      </div>
      <ul>
        <li
          className="spacer-bottom"
          key="cat1"
        >
          <HotspotCategory
            categoryKey="cat1"
            hotspots={
              Array [
                Object {
                  "author": "Developer 1",
                  "component": "com.github.kevinsawicki:http-request:com.github.kevinsawicki.http.HttpRequest",
                  "creationDate": "2013-05-13T17:55:39+0200",
                  "key": "h3",
                  "line": 81,
                  "message": "'3' is a magic number.",
                  "project": "com.github.kevinsawicki:http-request",
                  "resolution": undefined,
                  "rule": "checkstyle:com.puppycrawl.tools.checkstyle.checks.coding.MagicNumberCheck",
                  "securityCategory": "cat1",
                  "status": "TO_REVIEW",
                  "updateDate": "2013-05-13T17:55:39+0200",
                  "vulnerabilityProbability": "MEDIUM",
                },
                Object {
                  "author": "Developer 1",
                  "component": "com.github.kevinsawicki:http-request:com.github.kevinsawicki.http.HttpRequest",
                  "creationDate": "2013-05-13T17:55:39+0200",
                  "key": "h4",
                  "line": 81,
                  "message": "'3' is a magic number.",
                  "project": "com.github.kevinsawicki:http-request",
                  "resolution": undefined,
                  "rule": "checkstyle:com.puppycrawl.tools.checkstyle.checks.coding.MagicNumberCheck",
                  "securityCategory": "cat1",
                  "status": "TO_REVIEW",
                  "updateDate": "2013-05-13T17:55:39+0200",
                  "vulnerabilityProbability": "MEDIUM",
                },
              ]
            }
            onHotspotClick={[MockFunction]}
            onToggleExpand={[Function]}
            selectedHotspot={
              Object {
                "author": "Developer 1",
                "component": "com.github.kevinsawicki:http-request:com.github.kevinsawicki.http.HttpRequest",
                "creationDate": "2013-05-13T17:55:39+0200",
                "key": "h2",
                "line": 81,
                "message": "'3' is a magic number.",
                "project": "com.github.kevinsawicki:http-request",
                "resolution": undefined,
                "rule": "checkstyle:com.puppycrawl.tools.checkstyle.checks.coding.MagicNumberCheck",
                "securityCategory": "command-injection",
                "status": "TO_REVIEW",
                "updateDate": "2013-05-13T17:55:39+0200",
                "vulnerabilityProbability": "HIGH",
              }
            }
            title="cat1"
          />
        </li>
        <li
          className="spacer-bottom"
          key="cat2"
        >
          <HotspotCategory
            categoryKey="cat2"
            hotspots={
              Array [
                Object {
                  "author": "Developer 1",
                  "component": "com.github.kevinsawicki:http-request:com.github.kevinsawicki.http.HttpRequest",
                  "creationDate": "2013-05-13T17:55:39+0200",
                  "key": "h5",
                  "line": 81,
                  "message": "'3' is a magic number.",
                  "project": "com.github.kevinsawicki:http-request",
                  "resolution": undefined,
                  "rule": "checkstyle:com.puppycrawl.tools.checkstyle.checks.coding.MagicNumberCheck",
                  "securityCategory": "cat2",
                  "status": "TO_REVIEW",
                  "updateDate": "2013-05-13T17:55:39+0200",
                  "vulnerabilityProbability": "MEDIUM",
                },
              ]
            }
            onHotspotClick={[MockFunction]}
            onToggleExpand={[Function]}
            selectedHotspot={
              Object {
                "author": "Developer 1",
                "component": "com.github.kevinsawicki:http-request:com.github.kevinsawicki.http.HttpRequest",
                "creationDate": "2013-05-13T17:55:39+0200",
                "key": "h2",
                "line": 81,
                "message": "'3' is a magic number.",
                "project": "com.github.kevinsawicki:http-request",
                "resolution": undefined,
                "rule": "checkstyle:com.puppycrawl.tools.checkstyle.checks.coding.MagicNumberCheck",
                "securityCategory": "command-injection",
                "status": "TO_REVIEW",
                "updateDate": "2013-05-13T17:55:39+0200",
                "vulnerabilityProbability": "HIGH",
              }
            }
            title="cat2"
          />
        </li>
      </ul>
    </li>
  </ul>
  <ListFooter
    count={5}
    loadMore={[MockFunction]}
    loading={false}
  />
</div>
`;

exports[`should render correctly with hotspots: pagination 1`] = `
<div
  className="huge-spacer-bottom"
>
  <h1
    className="hotspot-list-header bordered-bottom"
  >
    <SecurityHotspotIcon
      className="spacer-right"
    />
    hotspots.list_title.TO_REVIEW.5
  </h1>
  <ul
    className="big-spacer-bottom"
  >
    <li
      className="big-spacer-bottom"
      key="HIGH"
    >
      <div
        className="hotspot-risk-header little-spacer-left"
      >
        <span>
          hotspots.risk_exposure
          :
        </span>
        <div
          className="hotspot-risk-badge spacer-left HIGH"
        >
          risk_exposure.HIGH
        </div>
      </div>
      <ul>
        <li
          className="spacer-bottom"
          key="cat2"
        >
          <HotspotCategory
            categoryKey="cat2"
            hotspots={
              Array [
                Object {
                  "author": "Developer 1",
                  "component": "com.github.kevinsawicki:http-request:com.github.kevinsawicki.http.HttpRequest",
                  "creationDate": "2013-05-13T17:55:39+0200",
                  "key": "h1",
                  "line": 81,
                  "message": "'3' is a magic number.",
                  "project": "com.github.kevinsawicki:http-request",
                  "resolution": undefined,
                  "rule": "checkstyle:com.puppycrawl.tools.checkstyle.checks.coding.MagicNumberCheck",
                  "securityCategory": "cat2",
                  "status": "TO_REVIEW",
                  "updateDate": "2013-05-13T17:55:39+0200",
                  "vulnerabilityProbability": "HIGH",
                },
              ]
            }
            onHotspotClick={[MockFunction]}
            onToggleExpand={[Function]}
            selectedHotspot={
              Object {
                "author": "Developer 1",
                "component": "com.github.kevinsawicki:http-request:com.github.kevinsawicki.http.HttpRequest",
                "creationDate": "2013-05-13T17:55:39+0200",
                "key": "h2",
                "line": 81,
                "message": "'3' is a magic number.",
                "project": "com.github.kevinsawicki:http-request",
                "resolution": undefined,
                "rule": "checkstyle:com.puppycrawl.tools.checkstyle.checks.coding.MagicNumberCheck",
                "securityCategory": "command-injection",
                "status": "TO_REVIEW",
                "updateDate": "2013-05-13T17:55:39+0200",
                "vulnerabilityProbability": "HIGH",
              }
            }
            title="cat2"
          />
        </li>
        <li
          className="spacer-bottom"
          key="cat1"
        >
          <HotspotCategory
            categoryKey="cat1"
            hotspots={
              Array [
                Object {
                  "author": "Developer 1",
                  "component": "com.github.kevinsawicki:http-request:com.github.kevinsawicki.http.HttpRequest",
                  "creationDate": "2013-05-13T17:55:39+0200",
                  "key": "h2",
                  "line": 81,
                  "message": "'3' is a magic number.",
                  "project": "com.github.kevinsawicki:http-request",
                  "resolution": undefined,
                  "rule": "checkstyle:com.puppycrawl.tools.checkstyle.checks.coding.MagicNumberCheck",
                  "securityCategory": "cat1",
                  "status": "TO_REVIEW",
                  "updateDate": "2013-05-13T17:55:39+0200",
                  "vulnerabilityProbability": "HIGH",
                },
              ]
            }
            onHotspotClick={[MockFunction]}
            onToggleExpand={[Function]}
            selectedHotspot={
              Object {
                "author": "Developer 1",
                "component": "com.github.kevinsawicki:http-request:com.github.kevinsawicki.http.HttpRequest",
                "creationDate": "2013-05-13T17:55:39+0200",
                "key": "h2",
                "line": 81,
                "message": "'3' is a magic number.",
                "project": "com.github.kevinsawicki:http-request",
                "resolution": undefined,
                "rule": "checkstyle:com.puppycrawl.tools.checkstyle.checks.coding.MagicNumberCheck",
                "securityCategory": "command-injection",
                "status": "TO_REVIEW",
                "updateDate": "2013-05-13T17:55:39+0200",
                "vulnerabilityProbability": "HIGH",
              }
            }
            title="cat1"
          />
        </li>
      </ul>
    </li>
    <li
      className="big-spacer-bottom"
      key="MEDIUM"
    >
      <div
        className="hotspot-risk-header little-spacer-left"
      >
        <span>
          hotspots.risk_exposure
          :
        </span>
        <div
          className="hotspot-risk-badge spacer-left MEDIUM"
        >
          risk_exposure.MEDIUM
        </div>
      </div>
      <ul>
        <li
          className="spacer-bottom"
          key="cat1"
        >
          <HotspotCategory
            categoryKey="cat1"
            hotspots={
              Array [
                Object {
                  "author": "Developer 1",
                  "component": "com.github.kevinsawicki:http-request:com.github.kevinsawicki.http.HttpRequest",
                  "creationDate": "2013-05-13T17:55:39+0200",
                  "key": "h3",
                  "line": 81,
                  "message": "'3' is a magic number.",
                  "project": "com.github.kevinsawicki:http-request",
                  "resolution": undefined,
                  "rule": "checkstyle:com.puppycrawl.tools.checkstyle.checks.coding.MagicNumberCheck",
                  "securityCategory": "cat1",
                  "status": "TO_REVIEW",
                  "updateDate": "2013-05-13T17:55:39+0200",
                  "vulnerabilityProbability": "MEDIUM",
                },
                Object {
                  "author": "Developer 1",
                  "component": "com.github.kevinsawicki:http-request:com.github.kevinsawicki.http.HttpRequest",
                  "creationDate": "2013-05-13T17:55:39+0200",
                  "key": "h4",
                  "line": 81,
                  "message": "'3' is a magic number.",
                  "project": "com.github.kevinsawicki:http-request",
                  "resolution": undefined,
                  "rule": "checkstyle:com.puppycrawl.tools.checkstyle.checks.coding.MagicNumberCheck",
                  "securityCategory": "cat1",
                  "status": "TO_REVIEW",
                  "updateDate": "2013-05-13T17:55:39+0200",
                  "vulnerabilityProbability": "MEDIUM",
                },
              ]
            }
            onHotspotClick={[MockFunction]}
            onToggleExpand={[Function]}
            selectedHotspot={
              Object {
                "author": "Developer 1",
                "component": "com.github.kevinsawicki:http-request:com.github.kevinsawicki.http.HttpRequest",
                "creationDate": "2013-05-13T17:55:39+0200",
                "key": "h2",
                "line": 81,
                "message": "'3' is a magic number.",
                "project": "com.github.kevinsawicki:http-request",
                "resolution": undefined,
                "rule": "checkstyle:com.puppycrawl.tools.checkstyle.checks.coding.MagicNumberCheck",
                "securityCategory": "command-injection",
                "status": "TO_REVIEW",
                "updateDate": "2013-05-13T17:55:39+0200",
                "vulnerabilityProbability": "HIGH",
              }
            }
            title="cat1"
          />
        </li>
        <li
          className="spacer-bottom"
          key="cat2"
        >
          <HotspotCategory
            categoryKey="cat2"
            hotspots={
              Array [
                Object {
                  "author": "Developer 1",
                  "component": "com.github.kevinsawicki:http-request:com.github.kevinsawicki.http.HttpRequest",
                  "creationDate": "2013-05-13T17:55:39+0200",
                  "key": "h5",
                  "line": 81,
                  "message": "'3' is a magic number.",
                  "project": "com.github.kevinsawicki:http-request",
                  "resolution": undefined,
                  "rule": "checkstyle:com.puppycrawl.tools.checkstyle.checks.coding.MagicNumberCheck",
                  "securityCategory": "cat2",
                  "status": "TO_REVIEW",
                  "updateDate": "2013-05-13T17:55:39+0200",
                  "vulnerabilityProbability": "MEDIUM",
                },
              ]
            }
            onHotspotClick={[MockFunction]}
            onToggleExpand={[Function]}
            selectedHotspot={
              Object {
                "author": "Developer 1",
                "component": "com.github.kevinsawicki:http-request:com.github.kevinsawicki.http.HttpRequest",
                "creationDate": "2013-05-13T17:55:39+0200",
                "key": "h2",
                "line": 81,
                "message": "'3' is a magic number.",
                "project": "com.github.kevinsawicki:http-request",
                "resolution": undefined,
                "rule": "checkstyle:com.puppycrawl.tools.checkstyle.checks.coding.MagicNumberCheck",
                "securityCategory": "command-injection",
                "status": "TO_REVIEW",
                "updateDate": "2013-05-13T17:55:39+0200",
                "vulnerabilityProbability": "HIGH",
              }
            }
            title="cat2"
          />
        </li>
      </ul>
    </li>
  </ul>
  <ListFooter
    count={5}
    loadMore={[MockFunction]}
    loading={false}
    total={7}
  />
</div>
`;
